### DISCUSSION TITLE
How does the IF activity in Elsa 3.1.3 designer define an expression where the condition evaluates to true or false?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5709] [General]
How does the IF activity in Elsa 3.1.3 designer define an expression where the condition evaluates to true or false?

**Comment by glime-ai** [comment]
In Elsa 3.1.3, the `If` activity allows you to define a condition that evaluates to true or false through the `Condition` attribute. When you use the `If` activity in the designer, you specify the condition as an expression. This expression is evaluated at runtime, and if the result is `true`, the workflow will proceed with the activities defined in the `Then` branch; otherwise, it will proceed with the activities in the `Else` branch.

The `Then` and `Else` branches are represented as sub-flowcharts within the `If` activity. In the designer interface, you will not see separate properties for `Then` and `Else`. Instead, you will see two buttons in the activity display that lead to sub-flowcharts where you can implement the logic for each branch by dragging and dropping new activities.

To set the condition, you would typically use an expression that can be evaluated to a boolean value. The exact syntax for the expression will depend on the expression evaluator that you are using within Elsa (e.g., JavaScript expressions, Liquid expressions, etc.).

For example, if you are using JavaScript expressions, you might define a condition like this:

```javascript
{{ someVariable > 10 }}
```

This expression checks if the value of `someVariable` is greater than 10, which will evaluate to either `true` or `false`.

Please note that as of the knowledge cutoff date, there have been reports that the `Then` and Else` properties are not visible in the designer interface for Elsa 3.1.3. However, the functionality is still accessible through the sub-flowcharts as described above.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
